### ISSUE TITLE
Support Flutter 3.19.0+ & Dart 3.0.0+

### ISSUE THREAD

**Issue by thibaultzanini** [onepub-dev/onepub#532]
OnePub CLI is broken with the latest version of Flutter(3.19.0+)/Dart(3.3.0)

** Problems with CLI tooling **

```
onepub doctor

Dart version:   3.3.0                              
Synchronous waiting using dart:cli waitFor and C API Dart_WaitForEvent is deprecated and disabled by default. This feature will be fully removed in Dart 3.4 release. You can currently still enable it by passing --enable_deprecated_wait_for to the Dart VM. See https://dartbug.com/52121.
dart:cli-patch/cli_patch.dart 9:47                                                 _waitForEvent
dart:cli/wait_for.dart 64:12                                                       _WaitForUtils.waitForEvent
dart:cli/wait_for.dart 152:19                                                      waitFor
package:dcli/src/util/wait_for_ex.dart 38:17                                       waitForEx
package:dcli/src/functions/which.dart 97:5                                         which
package:dcli/src/script/dart_sdk.dart 656:16                                       DartSdk._determineDartPath
package:dcli/src/script/dart_sdk.dart 40:39                                        DartSdk._pathToDartExe
package:dcli/src/script/dart_sdk.dart                                              DartSdk._pathToDartExe
package:dcli/src/script/dart_sdk.dart 78:32                                        DartSdk.pathToDartExe
package:onepub/src/commands/doctor.dart 174:38                                     _printPlatform
package:onepub/src/commands/doctor.dart 32:5                                       DoctorCommand.run
package:args/command_runner.dart 212:27                                            CommandRunner.runCommand
package:args/command_runner.dart 122:25                                            CommandRunner.run.<fn>
dart:async/future.dart 305:31                                                      new Future.sync
package:args/command_runner.dart 122:14                                            CommandRunner.run
package:onepub/src/entry_point.dart 38:20                                          entrypoint
/Users/thibaultzanini/.pub-cache/hosted/pub.dev/onepub-4.0.0/bin/onepub.dart 11:9  main
dart:isolate-patch/isolate_patch.dart 295:33                                       _delayEntrypointInvocation.<fn>
dart:isolate-patch/isolate_patch.dart 184:12                                       _RawReceivePort._handleMessage
package:dcli/src/util/wait_for_ex.dart 35:28                                       waitForEx
package:dcli/src/functions/which.dart 97:5                                         which
package:dcli/src/script/dart_sdk.dart 656:16                                       DartSdk._determineDartPath
package:dcli/src/script/dart_sdk.dart 40:39                                        DartSdk._pathToDartExe
package:dcli/src/script/dart_sdk.dart                                              DartSdk._pathToDartExe
package:dcli/src/script/dart_sdk.dart 78:32                                        DartSdk.pathToDartExe
package:onepub/src/commands/doctor.dart 174:38                                     _printPlatform
package:onepub/src/commands/doctor.dart 32:5                                       DoctorCommand.run
package:args/command_runner.dart 212:27                                            CommandRunner.runCommand
package:args/command_runner.dart 122:25                                            CommandRunner.run.<fn>
dart:async/future.dart 305:31                                                      new Future.sync
package:args/command_runner.dart 122:14                                            CommandRunner.run
package:onepub/src/entry_point.dart 38:20                                          entrypoint
/Users/thibaultzanini/.pub-cache/hosted/pub.dev/onepub-4.0.0/bin/onepub.dart 11:9  main
dart:isolate-patch/isolate_patch.dart 295:33                                       _delayEntrypointInvocation.<fn>
dart:isolate-patch/isolate_patch.dart 184:12                                       _RawReceivePort._handleMessage
```

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Version 14.3

**Comment by bsutton** [comment]
There is now a fix available.

To test the fix run:

dart pub global activate onepub 4.1.0-beta.1

please report any issues here.

If now problems emerge will release this to stable in the next couple of days.

**Comment by bsutton** [comment]
this has now been officially released to stable as 4.1.0

apologies for the delay.
